dataproc_jupyter_plugin/services/dataproc.py

```diff
@@ -52,27 +58,146 @@ async def list_clusters(self, page_size, page_token):
                     return resp
                 else:
                     return {
-                        "error": f"Failed to fetch clusters: {response.status} {await response.text()}"
+                        "error": f"Failed to fetch runtimes: {response.status} {await response.text()}"
                     }
+        except Exception as e:
+            self.log.exception(f"Error fetching runtime list: {str(e)}")
+            return {"error": str(e)}
+
+    async def list_clusters(self, page_size, page_token):
+        try:
+            # Create a client
+            client = dataproc.ClusterControllerAsyncClient(
+                client_options={
+                    "api_endpoint": f"us-central1-dataproc.googleapis.com:443"
+                },
+                credentials=oauth2.Credentials(self._access_token),
+            )
+
+            # Initialize request argument(s)
+            request = dataproc.ListClustersRequest(
+                project_id=self.project_id,
+                page_size=int(page_size),
+                page_token=page_token,
+                region=self.region_id,
+            )
+
+            # Make the request
+            page_result = await client.list_clusters(request=request)
+            clusters_list = []
 
+            # Handle the response
+            async for response in page_result:
+                clusters_list.append(json.loads(proto.Message.to_json(response)))
```

Comment:
You're traversing the message, serializing it as a string representing a JSON object, and then parsing that string to get back a Python dictionary.

All of the while, there is [a corresponding method](https://github.com/googleapis/proto-plus-python/blob/747aeddf0717b06ea9a226adead39899cb08cb50/proto/message.py#L552) that just directly generates a dictionary without having to write to a string first.

Further, you are taking these resulting dictionaries, which use integers for enum values, and manually converting those integers to the corresponding enum value names.

However, I see that there is a [keyword parameter](https://github.com/googleapis/proto-plus-python/blob/747aeddf0717b06ea9a226adead39899cb08cb50/proto/message.py#L556) on these methods that will use the enum value names to begin with if it is set to `False`.

Please change all of the calls to `proto.Message.to_json(...)` in this file to corresponding calls to `proto.Message.to_dict(..., use_integers_for_enums=False)`, and then delete the integer-to-enum value name mapping from the constants file.